crates/turborepo-lib/src/shim/mod.rs

```diff
@@ -75,7 +75,9 @@ fn run_correct_turbo(
     ui: ColorConfig,
 ) -> Result<i32, Error> {
     if let Some(turbo_state) = LocalTurboState::infer(&repo_state.root) {
-        try_check_for_updates(&shim_args, turbo_state.version());
+        let builder = crate::config::TurborepoConfigBuilder::new(&repo_state.root);
+        let config = builder.build().unwrap_or_default();
```

Comment:
We need to be careful here as there could be a delta between the configuration loaded this way and the actual configuration.

The only one I know of is `--root-turbo-json` which changes the location of the `turbo.json` used for configuration. Can you add that flag to the shim arg parser and use it if present in the config builder? Please add tests in the `shim/parser.rs` module.